patches/server/0738-Fix-entities-glitching-through-blocks-pushed-by-pist.patch

```diff
@@ -0,0 +1,80 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Peter Svensson <petersv@psv.nu>
+Date: Mon, 2 Aug 2021 12:06:05 +0200
+Subject: [PATCH] Fix entities glitching through blocks pushed by pistons
+ (#5189)
+
+Fix #5189 where entities in slime blocks are unreliable.
+
+Add a general mechanism where the entity knows how many ticks into
+the future that it has to be fully ticked. ActivationRange respects
+that and does not cull any of those ticks to the entity.
+
+diff --git a/src/main/java/net/minecraft/world/entity/Entity.java b/src/main/java/net/minecraft/world/entity/Entity.java
+index 896d892237b29eb404398db07264eb6f04786754..b16b51427b0d397b3e1b50d475bdf6ef5cd4e660 100644
+--- a/src/main/java/net/minecraft/world/entity/Entity.java
++++ b/src/main/java/net/minecraft/world/entity/Entity.java
+@@ -690,6 +690,11 @@ public abstract class Entity implements Nameable, EntityAccess, CommandSource, n
+ 
+         this.firstTick = false;
+         this.level.getProfiler().pop();
++        // Paper start - track when all ticks are required for the entity, e.g. after a piston push
++        if (this.isTickRequired()) {
++           this.remainingExactTicks -= 1;
++        }
++        // Paper end
+     }
+ 
+     public void setSharedFlagOnFire(boolean onFire) {
+@@ -3992,4 +3997,16 @@ public abstract class Entity implements Nameable, EntityAccess, CommandSource, n
+         return ((ServerChunkCache) level.getChunkSource()).isPositionTicking(this);
+     }
+     // Paper end
++ 
++    // Paper start - track when all ticks are required for the entity, e.g. after a piston push
++    private int remainingExactTicks = 0;
++
++    public void requestExactTicks(int ticks) {
++        this.remainingExactTicks = Math.max(ticks, this.remainingExactTicks);
++    }
++
++    public boolean isTickRequired() {
++        return (this.remainingExactTicks > 0);
++    }
++    // Peter end
```

Comment:
```suggestion
+    // Paper end
```

:P

Reply:
also, since there's already a paper block above this, you can just move this into the other paper block (i.e. remove lines 32-34)

Reply:
Updated in new commit